LPS-190069

ces-locpham opened the pull request:
Hi @wanderlast, 
I updated PR follow to the sheet which you mention me. I think it is no reason to remove it, so, I added again the deleted country and regions.

Commit:
LPS-198933 Remove service from PageElementDTOConverter, DTOConverter implementations must be registered as a DTOConverter service only

ling-alan-huang committed:
LPS-157776 Skip *-legacy-*.properties

Commit:
LPS-198120 Create new validation

Commit:
LPS-198120 Create new language key

Commit:
LPS-198120 buildLang

Kvale1733 committed:
LRQA-83527 Go to Server Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Instance Setting with url navigation

Kvale1733 committed:
LRQA-83527 Go to Users Admin with URL navigation

Kvale1733 committed:
LRQA-83527 Go to Sites Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Virtual Instance Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Accounts Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Workflows admin with url navigation

alessiorendina committed:
COMMERCE-12447 Added new method to inventory display contexts

alessiorendina committed:
COMMERCE-12447 Update usage

alessiorendina committed:
COMMERCE-12447 Updated data set dto

Kvale1733 committed:
LRQA-83527 Go to Roles Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Workflow Metrics Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Channels Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Warehouses Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Products Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Discounts admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Orders Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Custom Fields Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Price Lists Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Submissions Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Commerce Inventory Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to App Manager Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Audit Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Objects Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to User Monitoring Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Policies Admin with url navigation

Kvale1733 committed:
LRQA-83527 Go to Client Extensions Admin with url navigation

Commit:
LPS-198120 Fix test

gislayne-vitorino committed:
LPS-201940 (change-tracking-store-test) Add test for CTSContentTableReferenceDefinition

Kvale1733 committed:
LRQA-84215 Modifed macros to use correct creator email address for virtual instance

chestofwonders committed:
LPS-202135 Exchange flipping and style evaluation order

devpedrotavares committed:
LPS-196240 Only clear the ThreadLocal's ObjectEntryIdsMap when current service call is not itself product of an action. Also, clear before since the update action is done asynchronously

izaera committed:
LPS-202104 Make ${modifiedTimestamp} variable work

Add a substitution variable we can use in URL fields like this:

      /index.js?t=${modifiedTimestamp}

So that you can use it to force cache invalidation, or for any other
thing that may fit.

Note that its value comes from a buildTimestamp property in the case of
CXs deployed from workspace and from the DB's modifiedTime in the case of
CXs deployed from UI.

smottal committed:
COMMERCE-12839 Protect publisher menu item

izaera committed:
LPS-202104 Simplify how CET objects are created

Since we have added the notion of raw CET objects in the previous commit
and since we are putting more and more logic into each CET type
implementation I'm moving as much responsibility as possible to the
CETImplFactory classes and removing it from the CET implementation.

This:

1. Simplifies a lot the implementation of new types.
2. Centralizes operations in expected places.
3. And makes it easier to manage CET types without having to create CET
   instances just to ask them.

Note that we can dig deeper into 3 because there is still some code that
needs CET instantions but could be invoked without creating that.

I'll refactor that in further commits.

izaera committed:
LPS-202104 Simplify even more

By moving duplicated code from CETImplFactory implementations to
CETFactoryImpl we can simplify code more.

izaera committed:
LPS-202104 Create a proper CET given that it is possible

We haven't needed this until now because none of the unfilled fields were
read, but it's a matter of time that we end up having an issue because of
this, so let's build the objects properly since we can.

izaera committed:
LPS-202104 Use CET instead of ClientExtensionEntry whenever possible

Since the CETFactory can construct CETs from a PortletRequest, formalize
the contract between PortletRequest's parameters and the CETFactory and
stop using ClientExtensionEntry objects.

This simplifies code and makes using two objects (the CET and the
ClientExtensionEntry) to render the UI unnecessary.

izaera committed:
LPS-202104 Remove the raw CET concept

After the previous refactors, I've realized we don't really need a new
not very well defined concept like raw CET. We can simply move it to a
boolean parameter in the factory and it's the caller's responsibility to
make sure it doesn't use the incorrect CET version (raw vs those with
variables replaced).

The fact is that we don't mix raw and replaced CETs in any part of DXP, we
either use all raw or all replaced CETs, so we can do this new refactor.

izaera committed:
LPS-202104 SF

sandrodw3 committed:
LPS-201494 Add KeyboardMovementContext

devpedrotavares committed:
LPS-196240 Create integration tests to ensure actions can't form an infinite loop

izaera committed:
LPS-202104 semver

# breaking

## What modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/configuration/CETConfiguration.java

modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/configuration/CETConfiguration.java
now has a new field `buildTimestamp` that can be informed by Liferay
Workspace whenever a new build takes place.

## Why

DXP will use this field to set the value of the ${modifiedTimestamp}
substitution variable in client extensions.

----

# breaking

## What modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/factory/CETFactory.java

modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/factory/CETFactory.java
has changed the way it creates CET objects in two ways:

1. A new parameter `replaceVariables` to control variable substitution
   in URL fields when a CET is created by means of ClientExtensionEntry
   or CETConfiguration has been added.
2. For the case of creating a CET from a PortletRequest, there's no
   need to provide the `type` any more as it is inferred from
   PortletRequest's parameters.

## Why

This is part of a revamp of the API to make it easier to understand,
easier to maintain, and easier to extend.

----

# breaking

## What modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/factory/CETImplFactory.java

modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/factory/CETImplFactory.java
has changed its contract so that the whole responsibility of
constructing specific CET objects is moved to these factories, instead
of remaining buried in the CET implementation class.

See the changes in the interface itself for more information.

## Why

Some of the operations with specific CETs required to create an instance
for nothing simply because the code was in the CET implementation
instead of the factory. With this change that is no longer needed.

Additionally, the new contract allows for much simpler implementations
of CETImplFactory classes, because less code is needed.

----

# breaking

## What modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/CET.java

modules/apps/client-extension/client-extension-type-api/src/main/java/com/liferay/client/extension/type/CET.java
has a new method `getName()` that allows obtaining the full localization
XML that defines the name of the client extension.

## Why

Previously there was only a `getName(Locale)` method that allowed to
obtain the name in a given locale, but there are situations where we
need the whole thing (the complete XML with all labels for all locales).

----

izaera committed:
LPS-202104 Make site initializer extender test compile

smottal committed:
COMMERCE-12839 Supplier and Delivery API usage

smottal committed:
COMMERCE-12839 Fix companyId as string

smottal committed:
COMMERCE-12839 Auto SF

MirnaGama committed:
LPS-202614 Get tabs2 method will check for the role,  if it's an admin role then the segments section will be unavailable and users section will be requested instead

MirnaGama committed:
LPS-202614 Adjust methods that call for getTabs2

MirnaGama committed:
LPS-202614 Use method that checks the role instead

ling-alan-huang committed:
LPS-157776 LPS-157776 Group and sort properties, portal.properties | not documented | portal-osgi-env properties

ling-alan-huang committed:
LPS-157776 Rename

ling-alan-huang committed:
LPS-157776 Enforce an empty line before/after multiline values property

ling-alan-huang committed:
LPS-157776 Fix for egde case

see: https://github.com/liferay/liferay-portal/blob/d2b349495f5e94bb8a828fa3142ea01c8b85bc92/portal-impl/src/portal-liferay-online-config.properties

ling-alan-huang committed:
LPS-157776 Skip generated fils, see portal-instance.properties

ling-alan-huang committed:
LPS-157776 Check the existence of properties in portal.properties even it is commented out

ling-alan-huang committed:
LPS-157776 Rename

ling-alan-huang committed:
LPS-157776 Skip portal-osgi-configuration.properties as it is a generated file

ling-alan-huang committed:
LPS-157776 Fix typo

markomihalovics committed:
LPS-200885 Adjusted the logic in the "FriendlyURLServlet.getRedirect" method to maintain the value of the "redirectLayout" variable as "null" when a User lacks the permissions to view the requested page. This way, the 404 Page Not Found layout will be returned to the User.

markomihalovics committed:
LPS-200885 Code format in the "FriendlyURLServlet".

sandrodw3 committed:
LPS-201494 Add KeyboardMovementManager

sandrodw3 committed:
LPS-201494 Render manager

sandrodw3 committed:
LPS-201494 Enable movement when pressing enter on sidebar items

robertoDiaz committed:
LPS-202517 Add new method to interface to  handle image creation

robertoDiaz committed:
LPS-202517 Basic method Implementation

robertoDiaz committed:
LPS-202517 Create new class based in GetCompletionMVCResourceCommand (simple Copy and paste)

robertoDiaz committed:
LPS-202517 Create a new Base class and call the right client method in each resourceMVC class

ling-alan-huang committed:
LPS-157776 Add more common prefixes

ling-alan-huang committed:
LPS-157776 SF

ling-alan-huang committed:
LPS-157776 Rename

ling-alan-huang committed:
LPS-157776 SF

ling-alan-huang committed:
LPS-157776 We assume all properties in portal.properties start with a letter

ling-alan-huang committed:
LPS-157776 SF

alvarosaugarlr committed:
LPS-200601 solving WebSsoProfileIntegrationTest

MirnaGama committed:
LPS-202614 Remove segments item if it is an admin role

Commit:
LPS-198933 Fix dto.class.name from ProductVirtualSettingsDTOConverter

Commit:
LPS-198933 Change ProductVirtualSettingsDTOConverter to be DTOConverter

Commit:
LPS-198933 Fix dto.class.name from SkuSubscriptionConfigurationDTOConverter

Commit:
LPS-198933 Change SkuSubscriptionConfigurationDTOConverter reference to be DTOConverter

ces-locpham committed:
LPS-190069 Update countries follow the ISO-3166

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

robertoDiaz committed:
LPS-202516 Add a new form in AICreator to request for images

victorg1991 committed:
LPS-202276 Create LayoutStructureRulesHelper

ambrinchaudhary committed:
LPS-202516 Add height to the modal

ambrinchaudhary committed:
LPS-202516 Rename parameters

sandrodw3 committed:
LPS-201494 We don't need this dropzone

sandrodw3 committed:
LPS-201494 Handle indexes in a more intuitive way

ambrinchaudhary committed:
LPS-202516 Show the images

ambrinchaudhary committed:
LPS-202516 SF

ambrinchaudhary committed:
LPS-202517 Add new keys

ambrinchaudhary committed:
LPS-202517 buildLang

ambrinchaudhary committed:
LPS-202516 Change for a label

ambrinchaudhary committed:
LPS-202516 SF

sandrodw3 committed:
LPS-201494 Improve drag areas sizes

sandrodw3 committed:
LPS-201494 Style dropzones and rows when they are being targeted

smottal committed:
COMMERCE-12447 Fix quantity format in sku table

sandrodw3 committed:
LPS-201494 We are always passing them as params

robertoDiaz committed:
LPS-202516 Rename parameter

victorg1991 committed:
LPS-202276 Implement rule processing

victorg1991 committed:
LPS-202276 Do not render hidden layout structure items

victorg1991 committed:
LPS-202276 Show hidden layout structure items

sandrodw3 committed:
LPS-201494 Rename

22LucaGiannotta22 committed:
LPS-202614 Add functional test

ling-alan-huang committed:
LPS-157776 Rename

sandrodw3 committed:
LPS-201494 Move to utils

sandrodw3 committed:
LPS-201494 Types

sandrodw3 committed:
LPS-201494 Add function to drop in row

sandrodw3 committed:
LPS-201494 Add function to drop in group

sandrodw3 committed:
LPS-201494 Add functions to get an item and a group

sandrodw3 committed:
LPS-201494 Adapt drag handler so it's a button that also enables keyboard movement

sandrodw3 committed:
LPS-201494 Scroll into dropzone when it's target

sandrodw3 committed:
LPS-201494 Style sidebar items and rows when they are keyboard source

Marina-Gouveia committed:
LPS-192659 updating field names in journal content

devpedrotavares committed:
LPS-196240 Move ObjectActionThreadLocal to object-api to use in in test

Marina-Gouveia committed:
LPS-192659 SF

ling-alan-huang committed:
LPS-157776 Skip when there are duplicate keys

ling-alan-huang committed:
LPS-157776 SF

jkappler committed:
LPS-202992 Adds new interface to help with logic from sitemap

jkappler committed:
LPS-202992 Baseline

jkappler committed:
LPS-202992 Adds implementation

jkappler committed:
LPS-202992 Uses new helper to exclude the layout from the sitemap

robertoDiaz committed:
LPS-202516 Split in 2 methods

jkappler committed:
LPS-202992 Adds integration tests

ling-alan-huang committed:
LPS-157776 SF

ling-alan-huang committed:
LPS-157776 SF

victorg1991 committed:
LPS-202319 Add tests

jaimelr10 committed:
LPS-202825 Create React component to handle export settings

jaimelr10 committed:
LPS-202825 Use ExportSettings component on jsp file

jaimelr10 committed:
LPS-202825 Manage fields component and buttons behaviour using events

jaimelr10 committed:
LPS-202825 Add clay layout dependency to package.json

ambrinchaudhary committed:
LPS-202205 Keep a counter to calculate when it gets out the dropzone.
 When entering in child elements, dragleave event is always fired, there is no way to prevent it.

michaelhashimoto committed:
LRCI-3892 Remove unnecessary JobInformation component

michaelhashimoto committed:
LRCI-3892 Remove unnecessary JobBuilds component

Kvale1733 committed:
LPS-202894 Update Tomcat to 9.0.83

michaelhashimoto committed:
LRCI-3892 Remove unnecessary BuildInformation component

michaelhashimoto committed:
LRCI-3892 Remove unnecessary BuildRuns component

sandrodw3 committed:
LPS-201494 Add custom className to all dropzone elements

michaelhashimoto committed:
LRCI-3892 Remove unnecessary UpstreamBuilds component

michaelhashimoto committed:
LRCI-3892 Remove unnecessary Jobs component

michaelhashimoto committed:
LRCI-3892 Rename HomePage to JobQueuePage

michaelhashimoto committed:
LRCI-3892 Remove unnecessary JobQueue component

michaelhashimoto committed:
LRCI-3892 Add the missing git branches endpoint

sandrodw3 committed:
LPS-201494 Add and render KeyboardMovementPreview

michaelhashimoto committed:
LRCI-3892 Rename JobBuildPage to BuildPage

michaelhashimoto committed:
LRCI-3892 Align the server endpoints to the client endpoints

sandrodw3 committed:
LPS-201494 Add language keys

sandrodw3 committed:
LPS-201494 buildLang

sandrodw3 committed:
LPS-201494 Update tests and snapshots

michaelhashimoto committed:
LRCI-3892 Add Jethr0ContainerFluid component to get rid of duplication

michaelhashimoto committed:
LRCI-3892 Add parameters to the JobEntities in order to make the process more flexible

alvarosaugarlr committed:
LPS-203046 solving XMLSecurityTest

SylviaLuan committed:
LPS-203056 Update value to fit all condition

Commit:
LPS-138647 Initial checkin

AliciaGarciaGarcia committed:
LPS-202830 prepare system to take into account that the version can come on the request

Commit:
LPS-138647 Patch to escape url obtained from the request and inserted into the <a> tag

Commit:
LPS-138647 Apply

LuisOrtiz89 committed:
LPS-202977 Adding support to generate more than one temporary index at a given moment by giving thems different names automatically

LuisOrtiz89 committed:
LPS-202977 Updating tests accordingly due to the new names given to the temporary indexes

Commit:
LPS-196035 Fix typo

markomihalovics committed:
LPS-200885 Adjusted the FriendlyURLServletTest.testGetRedirectOnHiddenLayout method to expect a 404 response.

Commit:
LPS-138647 build-lib-versions

AliciaGarciaGarcia committed:
LPS-202830 set the InfoItemIdentifier.VERSION_LATEST as default

saramart81 committed:
COMMERCE-12447 Add new functional test

izaera committed:
LPS-202104 Put t parameter removal under a FF

This is so that, while everything is setup in Workspace and other parts of the
solution, DXP keeps working as before.

Once everything is merged, tested, and correctly announced, we will remove the
FF and the old functionality (appending an automagic t parameter).

jaimelr10 committed:
LPS-202431 Remove Include Headers checkbox

chestofwonders committed:
LPS-202135 Fix test to adapt to new lifecycle

saramart81 committed:
COMMERCE-12447 Add Functional test and test fix

marcosapmf committed:
LRAC-15133 allowing lower time periods for retention policy in uat

michaelhashimoto committed:
LRCI-3892 Change the job parameters JSONArray into a JSONObject

ling-alan-huang committed:
LPS-157776 Skip examples

shuyangzhou committed:
LPS-203202 Parallel running DB init and framework init.

brianchandotcom committed:
LPS-202516 Rename

brianchandotcom committed:
LPS-202516 auto SF

robertoDiaz committed:
LPS-202517 Rename Base class to fit pattern

ling-alan-huang committed:
LPS-157776 Auto SF

gaborlovasdotcom committed:
LRQA-84647 Update tests

kevenleone committed:
COMMERCE-12839 New Hooks for Accounts, Product and Catalog

kevenleone committed:
COMMERCE-12839 Update Utils

kevenleone committed:
COMMERCE-12839 Update Types

kevenleone committed:
COMMERCE-12839 Update APIs

kevenleone committed:
COMMERCE-12839 Improve

kevenleone committed:
COMMERCE-12839 SF

Commit:
LPS-203194 Remove no longer needed test CannotDisplayIncludeHeaderWithExportBlogPostingJOSNLFormat

Commit:
LPS-203205 Update export objectDefinition in json format instead of csv

Commit:
LPS-203204 Update ExportTaskResourceTest.testPostExportTask

Hijky committed:
LPS-203206 Update a macro to create an API endpoint via UI

Commit:
LPS-203200 Remove redundant parameter

Commit:
LPS-203200 Correct db name

Hijky committed:
LPS-203206 Update testcases to test fix to create API endpoint via UI

Commit:
LPS-203199 Modify cases to adapt new markup

saramart81 committed:
COMMERCE-12447 Test Fix

Jeremy-Ch committed:
LPS-203188 Use a suitable navigation

Jeremy-Ch committed:
LPS-203193 Use a suitable click function

JanisZhang committed:
LPS-199958 breaking changes amend

Hijky committed:
LPS-186045 Add a testcase to assert headless batch engine can export child organization

Commit:
LPS-203201 Modify case to adapt new markup

izaera committed:
LPS-203203 Fix NPE if CSP configuration has not been set

Some of the requests won't pass through the CSP filter and the CSP
configuration won't be stored in the request. For those, we set the nonce
to '' as if CSP hadn't been configured.

veroglez committed:
LPS-203218 Take into account the new filtering by recent and mine

peerkar committed:
LPS-202989 portal-search-rest-impl: Try to create a legacy summary and get the description first from it

joseabelenda committed:
ISSD-1999 Increase the size of the name column to 255 chatacters in OAuth2Application

sandrodw3 committed:
LPS-201494 Use KeyboardKey type instead of constants

brianchandotcom committed:
LPS-192659 SF

brianchandotcom committed:
asset-display-page-api 14.1.0 prep next

brianchandotcom committed:
portal-language-lang 1.0.209 prep next

sandrodw3 committed:
LPS-201494 Move ScreenReaderAnnouncer components to layout-js-components-web

brianchandotcom committed:
asset-display-page-service 4.0.49 prep next

brianchandotcom committed:
layout-page-template-admin-web 2.0.110 prep next

brianchandotcom committed:
layout-page-template-service 4.0.97 prep next

brianchandotcom committed:
LPS-138647 prep next

carolmariaabb committed:
LPS-203238 Move Auto Incremental object field to release FF

brianchandotcom committed:
LPS-202614 SF

brianchandotcom committed:
LPS-202977 Match var name _tempIndexCounter in the test

brianchandotcom committed:
LPS-202989 SF

MirnaGama committed:
LPS-202614 Avoid NPE

davisaints committed:
LPS-198320 Run single test

joseabelenda committed:
ISSD-1999 Regen

victorg1991 committed:
LPS-202276 Move to correct folder

p2kmgcl committed:
LPS-203257 Remove feature flag

Mikellorza committed:
LPS-203219 Convert to Release Feature Flag

davisaints committed:
LPS-203236 Update value of the iframe title

brianchandotcom committed:
LPS-202830 SF

brianchandotcom committed:
LPS-202104 look at the source

brianchandotcom committed:
LPS-202104 Rename

alessiorendina committed:
COMMERCE-12447 Sort

brianchandotcom committed:
LPS-202104 Rename

brianchandotcom committed:
LPS-202104 Rename

Commit:
LPS-203261 Add feature flag via custom.properties on ObjectFieldsAutoIncrement.testcase

brianchandotcom committed:
LPS-202276 SF

pyoo47 committed:
LRCI-3673 Simplify, and check for invalid redact tokens

brianchandotcom committed:
LPS-202517 This is a better / more consistent name after all

brianchandotcom committed:
LPS-198120 Move to a more appropriate method

kevenleone committed:
ISSD-1802 Add Buyer Permissions, Update Flows

susanchen3 committed:
LPS-203280 Remove from Database Paritioning testray component

susanchen3 committed:
LPS-203280 Add to acceptance layer

brittneyq committed:
LRCI-3926 Set parallel executor update to 4 hours if release tester

brianchandotcom committed:
LPS-196240 Wordsmith

brianchandotcom committed:
LRAC-15104 Wordsmith

brianchandotcom committed:
LPS-201494 Regen

kevenleone committed:
ISSD-1802 Update Page Definition

Luizcarlosqueiroz committed:
LPS-192637 Publish entry before reordering

Kvale1733 committed:
LRQA-84192 Enable retries

brianchandotcom committed:
LRCI-3673 prep next

Luizcarlosqueiroz committed:
LPS-190733 Set wait to page load before get title

ces-locpham committed:
LPS-190069 Update regions follow the ISO-3166

ces-locpham committed:
LPS-190069 Add an upgrade process for region
